components/device-mgt/org.wso2.carbon.device.mgt.core/src/test/java/org/wso2/carbon/device/mgt/core/cache/DeviceCacheManagerImplTest.java

```diff
@@ -0,0 +1,129 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.device.mgt.core.cache;
+
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.core.cache.impl.DeviceCacheManagerImpl;
+import org.wso2.carbon.device.mgt.core.common.BaseDeviceManagementTest;
+import org.wso2.carbon.device.mgt.core.common.TestDataHolder;
+import org.wso2.carbon.device.mgt.core.config.DeviceConfigurationManager;
+import org.wso2.carbon.device.mgt.core.config.DeviceManagementConfig;
+import org.wso2.carbon.device.mgt.core.config.cache.DeviceCacheConfiguration;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+    Unit tests for DeviceCacheManagerImpl
+ */
+public class DeviceCacheManagerImplTest extends BaseDeviceManagementTest {
+    private static final int NO_OF_DEVICES = 5;
+    private static final String UPDATE_NAME = "updatedName";
+    private static final String DEVICE_TYPE = "TEST_TYPE";
+    private static final String DEVICE_ID_PREFIX = "TEST-DEVICE-ID-";
+    private DeviceCacheManagerImpl deviceCacheManager;
+    private List<DeviceIdentifier> deviceIds = new ArrayList<>();
+    private List<Device> devices = new ArrayList<>();
+    private List<DeviceCacheKey> deviceCacheKeyList = new ArrayList<>();
+
+    @BeforeClass
+    public void init() throws DeviceManagementException, IOException {
+        DeviceConfigurationManager.getInstance().initConfig();
+        for (int i = 0; i < NO_OF_DEVICES; i++) {
+            deviceIds.add(new DeviceIdentifier(DEVICE_ID_PREFIX + i, DEVICE_TYPE));
+        }
+        devices = TestDataHolder.generateDummyDeviceData(this.deviceIds);
+        deviceCacheManager = Mockito.mock(DeviceCacheManagerImpl.class, Mockito.CALLS_REAL_METHODS);
+        DeviceCacheConfiguration configuration = new DeviceCacheConfiguration();
+        configuration.setEnabled(true);
+        DeviceManagementConfig config = DeviceConfigurationManager.getInstance().getDeviceManagementConfig();
+        config.setDeviceCacheConfiguration(configuration);
+        initializeCarbonContext();
+    }
+
+    private void initializeCarbonContext() throws IOException {
+        if (System.getProperty("carbon.home") == null) {
+            File file = new File("src/test/resources");
+            if (file.exists()) {
+                System.setProperty("carbon.home", file.getAbsolutePath());
```

Comment:
Duplicated method ?

Reply:
+1 fixed in 9ef622da4d64b16865314ac54c0af3d1e208cbcc